### ISSUE TITLE
Some glitches, questions and a request

### ISSUE THREAD

**Issue by mm47** [garybentley/quollwriter#17]
Glitches:

1. In the option "Use the following language for the spellchecker", if we uncheck the "Set as default language" box, it resets (re-checks) as soon as we close the Options tab.

2. After some time adding info to Quoll (text, notes, characters, chapters...) it lags and increases the memory usage from ~400MB to almost 2GB. Closing the project and opening again solves this.

3. The highlighting on a text is uneven.
![Captura de tela_2021-10-15_21-10-04](https://user-images.githubusercontent.com/28028426/138360410-3a49304b-15ab-4a12-8931-505861fa788f.png)

4. Even when we set the timer to 1 hour, clicking on the clock icon while in fullscreen mode makes the timer pops up every 30 minutes, which is very intrusive and distracting.

5. I started translating with QW version 3.0.0b16. Then I updated to 3.0.0b17 and when I hit the eye button to try out my translation, it tells me that I can't do that because it is a different version of Quoll.

6. On the translation window, even when we complete the sets, it still shows zero: 0/10, 0/33, etc. Closing the project and reopening solves it, but we have to do this every time.

7. Every once in a while my configuration is reset. I lose font config, colors and even achievements. All reset. I wasn't able to replicate this. But it has happened three times by now. Data is not lost though (phew!) just the software's config.

8. I'm unable to delete a warm-up exercise.

Questions:

1. Is there any way to disable the spellchecker? I write in another language that is not English.

2. What "to init" means? There's a string "Unable to init file."

3. What should I put in the place of a blue link? For example:
![Captura de tela_2021-10-21_14-51-12](https://user-images.githubusercontent.com/28028426/138360452-1fc84f89-6d83-4003-9ec6-c8f95d8a10ee.png)
![Captura de tela_2021-10-21_16-50-08](https://user-images.githubusercontent.com/28028426/138360456-adad7c05-70db-401e-9d5d-56d2d0634be2.png)

4. There's a missing {} (maybe a typo?) in one string:
"Warning! Quoll Writer has been unable to re-download the User Interface strings for your selected language. There may be multiple reasons for this, such as a connection error to the internet or that the Quoll Writer server is unavailable.

It is recommended that you either restart **Quoll Writer}** to try again or try downloading the strings from the Options panel.

In the interim Quoll Writer has fallen back to using English."

Request:

Pressing TAB to jump to the next text box on the translation window.

Edit: I'm using the Linux version. I have OpenJDK/JRE installed, versions 8, 11 and 17.

**Comment by garybentley** [comment]
Thanks for the report, I'm looking into the issues now.

For glitch 2, do you know what text you are adding that triggers this?  It might be an infinite loop.

**Comment by mm47** [comment]
No, it's normal text, like adding an Edit Needed Note; adding more text to Chapters (from LibreOffice and plain text); creating more characters and adding images and text, etc.

**Comment by garybentley** [comment]
Is the memory issue a problem that you can reproduce?  For example, does the same text cause it to start consuming memory every time?

**Comment by mm47** [comment]
No, not the same text. I add different amounts of text, different texts, but if I do this too much, the memory usage increases. If I want to add too much text, I have to close QW after a couple of chapters and notes, then reopen and continue. Maybe it has something to do with the fact that my chapters have around 20 thousand words? Maybe smaller chapters wouldn't spike the usage?

**Comment by garybentley** [comment]
So are you adding a lot of text quickly to a particular text box or are you doing it incrementally over a period of time or sessions?  It may be that the text areas have a memory bug.

**Comment by mm47** [comment]
Yes, incrementally on the same project. But if I open the project and only add a few paragraphs to a chapter or a couple of notes, nothing happens and I can leave Quoll open for hours. But if I add too much text quickly (let's say 50 notes with thousands words each from a plain text doc or libreoffice doc in a few minutes), than it increases the usage.
When I say notes, I mean the research notes. I never add more than a phrase to the Edit Needed notes.

**Comment by garybentley** [comment]
Ok thanks, that sounds like it's an issue with the text component.  I'll let the maintainer know.
